source/Octopus.Tentacle/Kubernetes/KubernetesClusterService.cs

```diff
@@ -13,15 +14,15 @@ public interface IKubernetesClusterService
     public class KubernetesClusterService : KubernetesService, IKubernetesClusterService
     {
         readonly AsyncLazy<ClusterVersion> lazyVersion;
-        public KubernetesClusterService(IKubernetesClientConfigProvider configProvider)
-            : base(configProvider)
+        public KubernetesClusterService(IKubernetesClientConfigProvider configProvider, ISystemLog log)
+            : base(configProvider, log)
         {
             //As the cluster version isn't going to change without restarting, we just cache the version in an AsyncLazy
             lazyVersion = new AsyncLazy<ClusterVersion>(async () =>
             {
                 var versionInfo = await Client.Version.GetCodeAsync();
                 return KubernetesVersionParser.ParseClusterVersion(versionInfo);
-            });
+            }, AsyncLazyFlags.RetryOnFailure);
```

Comment:
Lazy will cache the exception if you don't specify otherwise

Reply:
TIL